File: project/src/main/java/iteration2/project/NteCourse.java
package iteration2.project;

import java.util.HashMap;

public class NteCourse extends Course{

	public NteCourse(int courseSemester, String courseCode, String courseName, int credit, int akts, int quota,  String courseType) {
		super(courseSemester, courseCode, courseName, credit, akts, quota, courseType);
	}
	
	public boolean checkRequirement(Student student, String semester) {
		int noOfNTECourses = 0;
		for(Course course: student.getRequestedCourses()) {
			if(course.getCourseCode().equals(this.getCourseCode())) continue;
			if(course instanceof NteCourse) {
				noOfNTECourses++;
			}
			
		}
		if(semester.equals("FALL")) return false; // this course type can not be taken for FALL semester.
		
		// student can not take NTE courses more than 2.
		if(noOfNTECourses > 2) return false;
		
		return true;
		
	}
	

}

File: py_iteration/NteCourse.py
from Course import Course


class NteCourse(Course):
    def __init__(self, course_semester, course_code, course_name, credit, akts, quota, course_type):
        super().__init__(course_semester, course_code, course_name, credit, akts, quota, course_type)

    def check_course_requirement(self, **kwargs):
        accepted_courses = kwargs['accepted_courses']
        no_of_nte_courses = 0
        for course in accepted_courses:
            if type(course).__name__ == 'NteCourse':
                no_of_nte_courses += 1

        if no_of_nte_courses > 2:
            return False, 'can not take NTE courses more than 2.'
        return True, None